### DISCUSSION TITLE
Newbie questions (ESMira implementation)

### DISCUSSION THREAD

**Discussion by AFeneberg** [KL-Psychological-Methodology/ESMira#9] [Questions] [answered]
Hi, I've been trying out the platform (currently only the online demo version) and I am amazed by its potential. I have some questions regarding the implementation of ESMira and hope you can help out. 

Our IT support is able to implement ESMira on our web server, but we would need to perform regular updates (in case there are any) ourselves, so I was wondering how often you would recommend to check for necessary and perform updates? I assume any necessary updates from your side are posted here on Github?
The other question concerns the data storage, as far as I understood, the data are stored on the same server as the software. Would it be possible to separate these two, i.e., one server for front- and backend - a separate server for securing the data on a separate data-/fileserver behind a firewall, accessible only through the university network?

And a final research-oriented question: is it possible to implement dyadic ESM studies, e.g., two smartphones from two family memberes subscribed to the same study and as soon as they are close to each other (indicating a potential social interaction), a bluetooth connection checks this out and triggers a data entry? If you've already conducted such a study, I would highly appreciate a reference! 

Thank you very much in advance!

**Comment by JodliDev** [comment, marked as answer]
Hello
Regarding updates:
It really depends on what our focus is. Usually we try to be as back-portable as possible. But every now and then there are updates that are important. So I can't really think of a timetable. If you check every month or so, you should be fine. But in case you are not aware: ESMira is able to check and perform updates semi automatically: If you open the server settings, it automatically checks if there are new updates available and tells you. If so, you can tell it to automatically update

Regarding data storage:
Unfortunately that is not possible and I fear that is a bit out of scope for most use cases.

Regarding dyadic studies:
There is the bluetooth_list item that does something similar: it lists all BLE devices  (In anonymiued form) that are nearby and their distance. But it is implemented as a questionnaire item and is not triggered automatically. And it is also not limited to a specific device but aways lists all BLE devices it can find.
But I am also unsure if such a design is actually possible on smartphones. Android, and especially iOS, are quite strict on what is allowed to be done in the background. And measuring distances to Bluetooth devices is considered to be sensitive for privacy. I remember the Corona apps in 2021, which worked the same, needed specific API acces and I think that is still the case.

I hope that was helpful

**Comment by SelinaDev** [comment]
Hi!

Just to add to the data storage question: If you are able to configure your firewall to only allow access to certain subdomains it would theoretically be possible to install ESMira behind a firewall, but allow public access to certain API subdomains, e.g., those necessary to retrieve study information and those necessary for data upload, but disallow those subdomains necessary for data retrieval. However, this would mean no public access to the frontend, meaning no automatic study info pages on your server.
In case you are asking because of security reasons: The file system directory where data is saved can (and should) be separate from the server files. In the recommended configuration the data directory should not be accessible on its own via the internet (i.e., there is no `your.esmira.url/data` or similar that leads to the data). Under normal circumstances the data should only be accessible via the api. So if your firewall allows access of the (sub)domain ESMira runs on, the data are still separate (while of course we cannot make any guarantees regarding scurity riks of ESMira itself)

**Reply by AFeneberg** [reply]
Dear both, thank you very much for your helpful replies, this sounds awesome. Excited to set it up soon on our servers!
Best wishes
